File: src/runbot.php
<?php
require_once 'vendor/autoload.php';

use BotBattleClient\Runner;

/**
* Gets a command line argument
* @param int $index The index of the argument
* @param mixed $default The default value to return
*
* @return mixed The value passed for the gien argument or the default value
*/
function getArg(int $index, $default = null) {
    if ($_SERVER['argc'] > $index) {
        return $_SERVER['argv'][$index];
    }
    return $default;
}

if ($_SERVER['argc'] === 1) {
    echo "Usage:   ". $_SERVER['SCRIPT_FILENAME'] ." <username> [game_id] [difficulty] [bot_class]\n";
    echo "Example: ". $_SERVER['SCRIPT_FILENAME'] ." Bob null 1 RandomBot\n";
    exit(0);
}

// The API URL
$url        = 'http://10.0.0.10/api/';

// Get the passed settings
$username   = getArg(1, 'Random');
$gameId     = getArg(2, null);
$difficulty = getArg(3, 1);
$bot        = getArg(4, 'RandomBot');

$gameId = ($gameId !== null) && ($gameId !== "null") ? intval($gameId) : null;
$bot = "\\BotBattleClient\\Bots\\$bot";

// Start a bot runner
$runner = new Runner($url, $username, $difficulty, $bot, $gameId);
$runner->run();

File: src/python/runbot.py
import argparse

from battlebot.runner import Runner

if __name__ == '__main__':
    parser = argparse.ArgumentParser(description='Run a bot against the BotBattle game')
    parser.add_argument('username', help='The user to log in as')
    parser.add_argument('bot', default='RandomBot', nargs='?', help='The name of the bot class to run')
    parser.add_argument('game_id', default=None, nargs='?', help='The ID of the game to join (use -1 to find a game)')
    parser.add_argument('difficulty', default=1, nargs='?', help='The difficulty of game to join')

    args = parser.parse_args()
    if (args.username == None):
        parser.print_help()
        exit(0)

    api_url = 'http://10.0.0.10/api/'

    runner = Runner(base_url=api_url, username=args.username, bot=args.bot, game_id=args.game_id, difficulty=args.difficulty)
    runner.run()